### ISSUE TITLE
ApiGatewayToKinesisStreams Allows Setting IntegrationResponse but not MethodResponse

### ISSUE THREAD

**Issue by shellscape** [awslabs/aws-solutions-constructs#1130]
https://github.com/awslabs/aws-solutions-constructs/blob/8b30791902e09db2f7c49410a03d5d95ccc2ef51/source/patterns/%40aws-solutions-constructs/aws-apigateway-kinesisstreams/lib/index.ts#L179

This construct allows setting integration responses, but doesn't allow setting method responses, which must be done in tandem. The method response sets the expectation for the parameters that the integration response is allowed to send. Without both, setting responseParameter on the integration response is useless, and will always cause an error at deploy time. That error will resemble:

```
Resource handler returned message: "Invalid mapping expression specified: Validation Result: warnings : [], errors : [Invalid mapping expression parameter specified: method.response.header.Access-Control-Allow-Origin] (Service: ApiGateway, Status Code: 400, Request ID: cf213ad4-e67e-433f-afb8-39ec66a4bf42)" (RequestToken: f868444f-c9a5-79e1-09d4-656b26fd17dc, HandlerErrorCode: InvalidRequest), Resource handler returned message: "Invalid mapping expression specified: Validation Result: warnings : [], errors : [Invalid mapping expression parameter specified: method.response.header.Access-Control-Allow-Origin] (Service: ApiGateway, Status Code: 400, Request ID: 0e1f56d0-4d45-495b-a9c2-88de04f8d168)" (RequestToken: dfaab290-76bd-682f-37ff-7f2628196f44, HandlerErrorCode: InvalidRequest)
```

### Reproduction Steps

n/a

### Error Log

See above

### Environment

  - **CDK CLI Version      :** 2.141.0
  - **CDK Framework Version:** 2.141.0
  - **AWS Solutions Constructs Version     :** 2.58.1
  - **OS                   :** Mac
  - **Language             :** En

### Other

--- 

This is :bug: Bug Report

**Comment by biffgaut** [comment]
Thanks for the heads up, we'll take a look.

**Comment by biffgaut** [comment]
Can you provide a code example giving the error you're describing?

**Comment by biffgaut** [comment]
We're still trying to replicate your error to understand this issue better. The code below creates an Integration Response with as statusCode of 400 - it deploys successfully. It won't work, because there is no MethodResponse created that allows 400 to be returned to the client, but we're not seeing the deploy time error you describe above. It also includes mappings that introduce new attributes in the return JSON - these work for the 200 and 500 return values.

```
    new ApiGatewayToKinesisStreams(this, 'test-construct',  {
      putRecordIntegrationResponses: [
        {
          statusCode: '200',
          responseTemplates: {
            'application/json': "{\"kmsflag\": $input.json('$.EncryptionType')}"
          }
        },
        {
          statusCode: '500',
          selectionPattern: '500',
          responseTemplates: {
            'application/json': "{\"response\": $input.json('$.errorMessage')}"
          }
        },
        {
          statusCode: '400',
          selectionPattern: '400',
          responseTemplates: {
            'application/json': "{\"response\": $input.json('$.errorMessage')}"
          }
        }
      ]
    });
```

**Comment by biffgaut** [comment]
The fix to this issue has been published in v2.62.0
